h/views/oidc.py

```diff
@@ -149,36 +163,52 @@ def redirect(self):
 
         # Get the existing Hypothesis account that's already connected to this
         # ORCID iD, if any.
-        connected_user = self._user_service.fetch_by_identity(
-            IdentityProvider.ORCID, orcid_id
-        )
+        user = self._user_service.fetch_by_identity(IdentityProvider.ORCID, orcid_id)
 
         actions = {
             "connect": self.connect_orcid_id,
+            "login": self.log_in_with_orcid,
         }
         action_str = decoded_state["action"]
         action_method = actions[action_str]
-        return action_method(orcid_id, connected_user)
+        return action_method(orcid_id, user)
 
-    def connect_orcid_id(self, orcid_id: str, connected_user: User | None):
+    def connect_orcid_id(self, orcid_id: str, user: User | None):
         """Connect the user's ORCID iD to their Hypothesis account.
 
-        Do nothing if `orcid_id` is already connected to `connected_user`.
+        Do nothing if `orcid_id` is already connected to `user`.
 
         """
-        if connected_user and connected_user != self._request.user:
+        if user and user != self._request.user:
             # Oops, this ORCID iD is already connected to a different
             # Hypothesis account.
             raise UserConflictError
 
-        if not connected_user:
+        if not user:
             # This ORCID iD isn't connected to a Hypothesis account yet.
             # Let's go ahead and connect it to the user's account.
             self._orcid_client.add_identity(self._request.user, orcid_id)
 
         self._request.session.flash("ORCID iD connected ✓", "success")
         return HTTPFound(location=self._request.route_url("account"))
 
+    def log_in_with_orcid(self, orcid_id: str, user):
+        if not user:
+            msg = "Not implemented yet"
+            raise RuntimeError(msg)
+
+        del orcid_id
```

Comment:
`orcid_id` will be used once we add support for creating accounts that don't exist yet.

Reply:
This is an unused-parameter suppression?